projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/serverRequests.ts

```diff
@@ -47,6 +63,66 @@ type jobRequestResult = {
   isSuccessful: boolean;
 };
 
+type TaskStatusResult = {
+  /**
+   * Type of the task
+   */
+  task_type: string;
+  /**
+   * Status of the task
+   */
+  status: string;
+  /**
+   * Result message of the task if no errors occurred
+   */
+  message: string | IJobConvertToNotebookMessage;
+  /**
+   * Error message
+   */
+  error: string;
+};
+
+const pollForTaskCompletion = async (
+  taskId: string,
+  maxAttempts = 43200,
+  interval = 10000,
+  errorThreshold = 5
+): Promise<TaskStatusResult> => {
+  let errorCounter = 0;
+  for (let i = 0; i < maxAttempts; i++) {
+    try {
+      const result = await requestAPI<TaskStatusResult>(
+        `taskStatus?taskId=${taskId}`,
+        { method: 'GET' }
+      );
+      if (
+        result.task_type === taskId &&
+        (result.status === 'completed' || result.error)
+      ) {
+        return result;
+      }
+    } catch (error) {
+      showError(error);
+      errorCounter++;
+      if (errorCounter >= errorThreshold) {
+        return {
+          task_type: taskId,
+          status: 'error',
+          message: '',
+          error: `Error threshold (${errorThreshold}) reached. Task cancelled.`
+        };
+      }
+    }
+    await new Promise(resolve => setTimeout(resolve, interval));
+  }
+  return {
+    task_type: taskId,
+    status: 'error',
+    message: '',
+    error: 'Task timed out'
```

Comment:
Here we return that the task has timed out but what happens in the server, will the TaskRunner still be running the job itself or it will be terimated too?

Reply:
The timeout case is designed to happen after 12 hours pass without any success. This is also our limit for the VDK operations, right? If I am mistaken, correct me and I will add some server-side timeout mechanism as well.

Reply:
@yonitoo I am not sure about the limit of the VDK operations if it is the same, I suppose it is not a problem but i do think 12 hours is too much for this maybe @antoniivanov can share his opinion too? 12 h for UI imo too much maybe 5-6 h is better?